### ISSUE TITLE
Frequently asked questions

### ISSUE THREAD

**Issue by Corallo** [AxisCommunications/acap-computer-vision-sdk-examples#110]
This thread collets common questions that developers asked about our examples.

# Which cameras can I use to run these examples?
Only Artpec7 cameras equipped with TPU and Artpec8 cameras are supported at the moment.

# Can I use a different model?
Yes! You can use Ideally any model, however you have to make sure it is compatible with the hardware you want to run it on.

**On CPU,** you can run any model in the .tflite format.

**On EdgeTPU,** you have to verify that your model is [compatible](https://coral.ai/docs/edgetpu/models-intro) with the EdgeTPU the easiest way, is just to try to convert it to the EdgeTPU format with the [EdgeTPU compiler](https://coral.ai/docs/edgetpu/compiler/) and see the result. You can also check the [Supported operations](https://coral.ai/docs/edgetpu/models-intro/#supported-operations) when you build your model to see which operations can be used.

**On Artpec8,** you can run in theory any tflite model **quantized with int8 format**. We recommend that the model is quantized by tensor and not by channel to obtain better performance. If you are interested, take a look at [our guide](https://github.com/AxisCommunications/acap-native-sdk-examples/tree/master/tensorflow-to-larod-artpec8) about how to train and quantize a model for Artpec8. If some parts of your models can't be executed by the DLPU accelerator, they will be automatically sent to the CPU, this flexibility will come with a drop in performance, if you use a model that jumps between DLPU and CPU too much, the execution will be very slow. There is also a limit in how many times the inference can be handed over from the DLPU to the CPU which is currently set to 16.

# Can I run models in other formats like ONNX or PyTorch?
Unfortunately no. You could install the pytorch or onnx pip packages in your application, but these libraries won't have access to the hardware accelerators, and will run your model only on CPU. 

# SSD mobilenet performances are not good enough, can I run EfficientDet or YOLOv5 to improve the accuracy of the model? 
Yes, however, these two model are heavier than SSD mobilenetv2.
## EfficinetDet
[Some users](https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/discussions/108) successfully used EfficientDet Lite0 from the [coral website](https://coral.ai/models/object-detection/) on Artpec8 obtaining an inference time of 360ms.

## YOLOv5
We also saw that it is possible to use the popular YOLOv5, we recommend using the official implementation from the [Ultralytics repository](https://github.com/ultralytics/yolov5). In their repository, you can find also a script to convert their model to tflite (make sure to use the -int8 flag) and edgeTPU. Be aware that they quantize their model by channel, which is not the right quantization technique to get the best performance out of Artpec8 cameras.

We have collected some results from tests done with the YOLOv5 model
On Artpec8 using the "small" version of YOLOv5 and input size 640x640 the inference takes 1200 ms 
On Artpec7 using the "nano" version of YOLOv5 and input size 224x224 the inference takes 30-40 ms.

# Ask your question
If you didn't find what you were looking for, feel free to open a new thread in the discussion tab!
